pkg/process/runner/exclusive.go

```diff
@@ -0,0 +1,95 @@
+package runner
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/leptonai/gpud/pkg/log"
+	"github.com/leptonai/gpud/pkg/process"
+)
+
+var _ Runner = &exclusiveRunner{}
+
+func NewExclusiveRunner() Runner {
+	return &exclusiveRunner{}
+}
+
+// exclusiveRunner is a scheduler that runs a single process at a time.
+// Does not support concurrent execution of multiple processes.
+type exclusiveRunner struct {
+	mu      sync.RWMutex
+	running process.Process
+}
+
+var defaultScriptsDir = filepath.Join(os.TempDir(), "gpud-scripts-runner")
```

Comment:
shall we move it to the process pckage? IMO all these scripts and related files should be stored there. and I didn't see how to clean it.

Reply:
Sure, let me move

Reply:
@eahydra done